packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -156,15 +172,100 @@ function TokenModal({
     }
   }
 
+  const tokenExplorerUrl = buildExplorerTokenURL({
+    chainId: chain!.id,
+    tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+  });
+
   useEffect(() => {
     if (!internalBalanceData?.user?.userInternalBalances) return;
     getWalletBalance(tokenAdresses as `0x${string}`[]);
   }, [internalBalanceData]);
 
+  useEffect(() => {
+    if (addressRegex.test(tokenSearch)) {
+      setIsTokenSearchDisabled(false);
+      if (
+        tokens.filter((token) => filterTokenInput({ tokenSearch, token }))
+          .length < 1
+      ) {
+        fetchSingleTokenBalance({
+          tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+        }).then((tokenData) => {
+          setTokens((prev) => [
+            ...prev,
+            {
+              ...tokenData,
+              name: tokenData.symbol,
+              tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+            },
+          ]);
+        });
+      }
+    } else {
+      setIsTokenSearchDisabled(true);
+    }
+  }, [tokenSearch]);
+
+  function filterTokenInput({
+    tokenSearch,
+    token,
+  }: {
+    tokenSearch: string;
+    token?: TokenItem;
+  }) {
+    {
+      if (tokenSearch === "") return true;
+      const searchString = tokenSearch.toLowerCase();
+      if (token?.name?.toLowerCase().includes(searchString)) return true;
+      if (token?.symbol.toLowerCase().includes(searchString)) return true;
+      if (token?.tokenAddress.toLowerCase().includes(searchString)) return true;
+      return false;
+    }
+  }
+
   return (
     <div className="text-white divide-y divide-gray-700 max-h-[30rem] overflow-y-scroll scrollbar-thin scrollbar-thumb-gray-200 scrollbar-track-blue3">
       <div className="w-full flex flex-col justify-center items-center h-full py-4 gap-y-4">
         <div className="text-xl">Token Search</div>
+        <div className="w-full px-10 flex items-center">
+          <input
```

Comment:
Any reason not to use the standard input we have?

Reply:
They have different designs, we have to refactor the Input to accept variants 

https://linear.app/bleu-llc/issue/BAL-381/input-style-variants